ethereum/contracts/SimplifiedMMRVerification.sol

```diff
@@ -0,0 +1,81 @@
+pragma solidity ^0.8.5;
+
+contract  SimplifiedMMRVerification {
+    function verifyInclusionProof(
+        bytes32 root,
+        bytes32 leafNodeHash,
+        bytes32[] memory restOfThePeaks,
+        bool hasRightBaggedPeak,
+        bytes32 rightBaggedPeak,
+        bytes32[] memory merkleProofItems,
+        bool[] memory merkleProofOrder
+    ) public pure returns (bool) {
+        require(merkleProofOrder.length == merkleProofItems.length);
+        // TODO: Do validation of other input parameters
+
+        uint numberOfPeaks = 1 + restOfThePeaks.length;
+        if (hasRightBaggedPeak) {
+            numberOfPeaks++;
+        }
+
+        bytes32[] memory reversedPeaks = new bytes32[](numberOfPeaks);
+        uint peakInsertionPointer = 0;
+
+        if (hasRightBaggedPeak) {
+            reversedPeaks[peakInsertionPointer++] = rightBaggedPeak;
+        }
+
+        bytes32 merkleRootPeak = calculateMerkleRoot(leafNodeHash, merkleProofItems, merkleProofOrder);
```

Comment:
You should be able to user our existing MerkleProof.sol library for this

Reply:
@musnit The function `computeRootFromProofAtPosition` in the library is less optimized. It still uses pos and width, while we do not need it.

Reply:
Yeh - though pos and width may be less bytes than merkleProofOrder, so maybe the input calldata is slightly smaller.

I'm not sure which results in an effectively cheaper gas price, though I suspect the extra compute in the library is cheaper than the extra calldata that may be used for the merkleProofOrder.

Anyway, don't worry about this, I think the difference is probably fairly insignificant anyway.

Reply:
@musnit What do you think about bitfield? i.e we can have a uint256

Reply:
yeh true, maybe even uint16 or uint8 will be fine for a MMR for the first year or two while it has less than a billion ?

Reply:
I am using 64 bit bitfield now and it is tested. The reason behind 64 bit is rust code defines maximum height possible to be 63.